src/ct_selftest/ct_selftest.cpp

```diff
@@ -0,0 +1,297 @@
+/*
+ * Basic tests for the CT::poison annotations.
+ * Some of those are expected to fail, because they deliberately
+ * branch on secret memory.
+ *
+ * (C) 2024 Jack Lloyd
+ * (C) 2024 Fabian Albert, René Meusel - Rohde & Schwarz Cybersecurity
+ *
+ * Botan is released under the Simplified BSD License (see license.txt)
+ */
+
+#include <iostream>
+
+#include <botan/internal/ct_utils.h>
+
+#if defined(BOTAN_CT_POISON_ENABLED)
+
+   #include <botan/auto_rng.h>
+   #include <botan/hex.h>
+   #include <botan/internal/fmt.h>
+
+   #include <functional>
+   #include <map>
+
+namespace {
+
+void test_conditional_jump_on_poisoned_data() {
+   Botan::AutoSeeded_RNG rng;
```

Comment:
Let’s use System_RNG here - it should anyway be faster, and (hypothetically) some tools may have special understanding of `getentropy` etc

Reply:
Done. Instantiation moved to `main()`, as suggested in  https://github.com/randombit/botan/pull/4182#discussion_r1671811179